File: Java/src/main/GUI1_counter.java
package main;

import javafx.application.Application;
import javafx.geometry.Insets;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.TextField;
import javafx.scene.layout.HBox;
import javafx.stage.Stage;

public class GUI1_counter extends Application {

    public void start(Stage stage) {
        TextField count = new TextField("0");
        count.setEditable(false);
        count.setPrefWidth(50);
        Button countUp = new Button("Count up");
        Button countDown = new Button("Count down");

        countUp.setOnAction(e ->
                count.setText(Integer.parseInt(count.getText()) + 1 + ""));
        countDown.setOnAction(e ->
                count.setText(Integer.parseInt(count.getText()) - 1 + ""));

        HBox root = new HBox(20, countDown, count, countUp);
        root.setPadding(new Insets(25));

        stage.setScene(new Scene(root));
        stage.setTitle("Counter");
        stage.show();
    }

    public static void main(String[] args) {
        launch(args);
    }
}

File: Kotlin/src/main/kotlin/GUI1_Counter.kt
import javafx.geometry.Insets
import tornadofx.*

class GUI1_Counter : View("Counter") {
    override val root = borderpane {
        padding = Insets(10.0)
        center {
            hbox {
                spacing = 10.0
                val field = textfield {
                    text = "0"
                    prefColumnCount = 8
                    isEditable = false
                }
                button {
                    text = "Count"
                    setOnAction { field.text = "${field.text.toInt() + 1}"}
                }
            }
        }
    }
}